cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -1311,7 +1313,10 @@ public Secret generateBrokersSecret() {
             data.put(KafkaCluster.kafkaPodName(cluster, i) + ".p12", cert.keyStoreAsBase64String());
             data.put(KafkaCluster.kafkaPodName(cluster, i) + ".password", cert.storePasswordAsBase64String());
         }
-        return createSecret(KafkaCluster.brokersSecretName(cluster), data);
+        Map<String, String> annotations = Map.of(
+                clusterCa.caCertThumbprintAnnotation(), clusterCa.currentCaCertThumbprint(),
+                clientsCa.caCertThumbprintAnnotation(), clientsCa.currentCaCertThumbprint());
+        return createSecret(KafkaCluster.brokersSecretName(cluster), data, annotations);
```

Comment:
We seem to be adding the annotation here, for Zoo and for CC. But not for EO or for KE. Is the implementation of the secrets different there? Or why don't we need to do it?

Reply:
EO and KE use the `ModeUtils#buildSecret` method to build the Secret and it's updated to add the annotations.